keps/sig-node/2535-ensure-secret-pulled-images/README.md

```diff
@@ -382,58 +438,58 @@ container runtime. This KEP expects the image IDs provided to be in the form of
 type ImagePullIntent struct {
     metav1.TypeMeta
 
-    ImageSpec string `json:"imageSpec"` // the filename is a hash of this value
-    Credentials ImagePullCredentials `json:"credentials"`
+    // Image is the image spec from a Container's `image` field.
+    // The filename is a SHA-256 hash of this value. This is to avoid filename-unsafe
+    // characters like ':' and '/'.
+    Image string `json:"image"`
 }
 
+// ImagePullRecord is a record of an image that was pulled by the kubelet.
+//
+// If there are no records in the `kubernetesSecrets` field and both `nodeWideCredentials`
+// and `anonymous` are `false`, credentials must be re-checked the next time an
+// image represented by this record is being requested.
 type ImagePulledRecord struct {
     metav1.TypeMeta
 
-    ImageID string `json:"imageId"` // the filename is a filesystem sanitized image of this value
-    Credentials ImagePullCredentials `json:"credentials"`
-}
-
-// ImagePullCredentials serves two indexes
-// - First index serves for searching for a specific Kubernetes secret that contained
-//   the credentials used to pull an image. This assumes uninterrupted pulls if
-//   credentials were rotated in a Kubernetes secret.
-// - Second index serves for searching for a specific hash of credentials used to pull
-//   an image. This assumes uninterrupted pulls if the same set of credentials is
-//   spread across the cluster.
-type ImagePullCredentials struct {
-    // CredentialHashes is an index of hashes of all the credentials that an image was pulled with
-    // +optional
-    CredentialHashes map[string]ImagePullCredentials `json:"credentialHashes"`
+    // ImageRef is a reference to the image represented by this file as received
+    // from the CRI.
+    // The filename is a filesystem-sanitized form of this value.
+    ImageRef string `json:"imageRef"`
 
     // KuberneteSecretCoordinates is an index of coordinates of all the kubernetes
     // secrets that were used to pull the image
     // +optional
-    KubernetesSecretCoordinates map[KubernetesSecretCoordinates]ImagePullSecret `json:"kubernetesSecretCoordinates"`
+    KubernetesSecrets []ImagePullSecret `json:"kubernetesSecretCoordinates"`
 
     // NodeWideCredentials is a flag denoting that the image was pulled with node-wide
-    // credentials
+    // credentials.
+    // If true, it is mutually exclusive with Anonymous and KubernetesSecrets.
     // +optional
-    NodeWideCredentials bool `json:"nodeWideCredentials"`
+    NodeWideCredentials bool `json:"nodeWideCredentials,omitempty"`
 
     // Anonymous is a flag denoting that the image was pulled without authentication.
-    // It is mutually exclusive to all the other fields of this structure.
+    // If true, it is mutually exclusive with KubernetesSecrets and NodeWideCredentials.
     // +optional
-    Anonymous bool `json:"anonymous"`
+    Anonymous bool `json:"anonymous,omitempty"`
```

Comment:
do we need to differentiate between these?

Reply:
this is more of a cosmetic change, the deserialization stays the same